games/a/pommerman/envs/v0.py

```diff
@@ -501,20 +499,44 @@ def featurize(obs):
     def get_json_info(self):
         """Returns a json snapshot of the current game state."""
         ret = {
-            'attrs': {
-                'game_type': self._game_type,
                 'board_size': self._board_size,
-                'agent_view_size': self._agent_view_size,
-                'max_steps': self._max_steps,
-                'is_partially_observable': self._is_partially_observable,
-            },
-        }
-
-        for attr in ['training_agent', '_board', '_agents', '_bombs', '_flames', '_items', '_step_count']:
-            ret[attr] = utility.to_json(getattr(self, attr))
+                'step_count': self._step_count,
+                'board': utility.to_json(self._board),
+                'agents': utility.to_json(self._agents),
+                'bombs': utility.to_json(self._bombs),
+                'flames': utility.to_json(self._flames),
+                'items': utility.to_json(str(self._items))
+            }
+
         return ret
 
     def set_json_info(self):
         """Sets the game state as the init_game_state."""
-        # TODO
-        pass
+        self._step_count = self._init_game_state['step_count']
+        self._board_size = self._init_game_state['board_size']
+
+        boardArray = json.loads(self._init_game_state['board'])
+        self._board = np.ones((self._board_size, self._board_size)).astype(np.uint8) * utility.Item.Passage.value
+        for x in range(self._board_size):
+            for y in range(self._board_size):
+                self._board[x,y] = boardArray[x][y]
+
+        self._items = eval(json.loads(self._init_game_state['items']))
```

Comment:
why is this eval?

Reply:
Dictionary with int key tuple don't serialize well. I wrap the all 'items' section into a string and reverse it with eval, but with hindsight sounds quite hackish